design/dedicated-osd-pod.md

```diff
@@ -23,22 +23,34 @@ A more comprehensive discussions can be found at [this issue](https://github.com
 
 ## Proposal   
 
-We propose the following OSD startup change.
+We propose the following change to address the limitations.
+
+### Terms
+
+- OSD Discovery. A process that is given a device or directory path filter and use the filter to discover devices on the host. The process updates the OSD orchestration Configmap with the device paths and mark the orchestration status as "DeviceDiscovered". The process is created by Rook Operator and runs in a Kubernetes deployment on the storage node. This process is storage backend agnostic. Currently `rook osd` command implements this functionality.
+
+- OSD Provisioner. A process that is given device or directory paths upon start and make backend specific storage types. For now, the process prepare Ceph OSDs. Currently `rook osd` command implements this functionality.
 
 ### Create new OSDs
-| Sequence |Rook Operator  | OSD Provisioner   | Ceph OSD Deployment | 
-|---|---|---|---|
-| 1  | Read devices from cluster CRD and create a Pod for OSD Provisioner for each storage node  |   |   |
-| 2  | Watch Configmap  | Prepare OSDs  |   |
-| 3  | Watch Configmap  |  Persist OSD ID, datapath, and node info in a per node Configmap | |
-| 4  | Detect Configmap change, parse Configmap, extract OSD info, construct OSD Pod command and arg | |
-| 5  | Create one deployment per OSD | |
-| 6  | | | Start `ceph-osd` Daemon one Pod per device |
+| Sequence |Rook Operator | OSD Discovery  | OSD Provisioner   | Ceph OSD Deployment | 
+|---|---|---|---|---|
+| 1  | Read devices from cluster CRD and create an OSD Discovery deployment on each storage node  |   |   |
+| 2  | Watch Configmap  | Discover OSD devices that fit those defined in Cluster CRD and update Configmap with the device paths| | |
```

Comment:
I still believe initial device discovery should be assumed before cluster crd creation. And we already have an agent running everywhere intended for node configuration like this so I still don't see the need for a separate pod for device discovery.

Reply:
ok i see now your comment below about the agent.